ssg/build_yaml.py

```diff
@@ -335,64 +326,141 @@ def __sub__(self, other):
         return profile
 
 
-class ResolvableProfile(Profile):
+class ProfileWithSeparatePolicies(Profile):
     def __init__(self, * args, ** kwargs):
-        super(ResolvableProfile, self).__init__(* args, ** kwargs)
-        self.resolved = False
+        super(ProfileWithSeparatePolicies, self).__init__(* args, ** kwargs)
+        self.policies = {}
+
+    def read_yaml_contents(self, yaml_contents):
+        policies = yaml_contents.pop("policies", None)
+        if policies:
+            self._parse_policies(policies)
+        super(ProfileWithSeparatePolicies, self).read_yaml_contents(yaml_contents)
+
+    def _parse_policies(self, policies_yaml):
+        for item in policies_yaml:
+            id_ = required_key(item, "id")
+            controls = required_key(item, "controls")
+            if not isinstance(controls, list):
+                if controls != "all":
+                    msg = ("Policy {id_} contains invalid controls list {controls}."
+                        .format(id_=id_, controls=str(controls))
+                        )
+                    raise ValueError(msg)
+            self.policies[id_] = controls
 
     def resolve_controls(self, controls_manager):
         for policy_id, controls in self.policies.items():
+            to_expand = Control()
+
             if isinstance(controls, list):
-                controls_list = [controls_manager.get_control(policy_id, control_id) for control_id in controls]
+                to_expand = self._controls_to_items(controls_manager, policy_id, controls)
```

Comment:
`_controls_to_items` is defined in `ResolvableProfile`, but not here and not in `Profile`.

Reply:
Good point, if those special profile classes build upon `ResolvableProfile`, it will streamline things.